frontend/src/components/pages/home-page.tsx

```diff
@@ -277,25 +275,21 @@ const Node = ({ node, style }: NodeRendererProps<FileInfo>) => {
       );
     }
 
-    const relativePath =
-      node.data.path.startsWith(root) && Paths.isAbsolute(node.data.path)
-        ? node.data.path.slice(root.length + 1)
```

Comment:
Why was this slicing added? Is it okay to remove?

Reply:
This may just be masking a real issue. Relative paths should be ok to pass - this may happen from the sidebar as well. 

Why is passing a relative path returning a bad file?

Reply:
I think the real fix is on the backend that the relative path should use the directory from marimo edit, not the cwd

Reply:
Yea makes some sense.

Two reasons it returns a bad file.

1 is that the slicing logic can be off-by-one (which we can fix).

Other reason is that `AppFileRouter` doesn't use the directory, which I guess we can fix (can check if there is a directory?): https://github.com/marimo-team/marimo/blob/aa19fe6212e2bb8039e81b021ae26024d572e5df/marimo/_server/file_router.py#L73-L99

Reply:
Yea that's right. It's probably not wired down - I'm happy to do this Monday (I initially added the 'marimo edit dir' functionally). 

Also it's nice having this on the backend since it's much easier to test this

Reply:
Okay sounds good, I'll hold off. Thanks!

Reply:
Had spare time, so went ahead and did the change. Feel free to take over or revert if you had something else in mind.